### DISCUSSION TITLE
Installation of GaNDLF

### DISCUSSION THREAD

**Discussion by MaimoonaAkram** [mlcommons/GaNDLF#688] [General]
Actually I want to use google colab to do the task. I have installed conda on google colab conda 23.1.0, then I installed pytorch before using GaNDLF
but when I tried the following 
code it does not  work:
(base) $> conda create -n venv_gandlf python=3.8 -y
(base) $> conda activate venv_gandlf
(venv_gandlf) $> ### subsequent commands go here
## CUDA 11.6
# (venv_gandlf) $> pip install torch==1.13.1+cu116 torchvision==0.14.1+cu116 torchaudio==0.13.1 --extra-index-url https://download.pytorch.org/whl/cu116
## ROCm
# (venv_gandlf) $> pip install torch==1.13.1+rocm5.2 torchvision==0.14.1+rocm5.2 torchaudio==0.13.1 --extra-index-url https://download.pytorch.org/whl/rocm5.2
## CPU-only
# (venv_gandlf) $> pip install torch==1.13.1+cpu

**Comment by sarthakpati** [comment]
Hi @MaimoonaAkram,

Thanks for your interest in GaNDLF! 

I am a bit confused regarding which commands you tried. Can you give the step-by-step commands, please? For example, this is the command to create a new virtual environment:
```bash
conda create -n venv_gandlf python=3.8 -y
```

And this will activate it:
```bash
conda activate venv_gandlf
```

This command will install the correct version of PyTorch for CUDA hardware:

```bash
pip install torch==1.13.1+cu116 torchvision==0.14.1+cu116 torchaudio==0.13.1 --extra-index-url https://download.pytorch.org/whl/cu116
```

**Reply by MaimoonaAkram** [reply]
not working. plz screenshot given below:
![image](https://github.com/mlcommons/GaNDLF/assets/138813544/8b814231-ec9c-4de6-972b-7e0c082e9d37)

**Reply by sarthakpati** [reply]
It appears that you do not have a conda environment initialized. Have you created the base virtual environment in the correct manner? Though I don't use Google Colab at all, a quick internet search pointed me to **2** relevant answers regarding usage of Google Colab:

1. [This answer](https://towardsdatascience.com/conda-google-colab-75f7c867a522) explains how to install conda in a Google Colab environment.
1. [This answer](https://stackoverflow.com/a/53777433) explains how to install PyTorch in a Google Colab environment.

**Comment by MaimoonaAkram** [comment]
Can I run your GaNDLF framework on google COLAB??

**Reply by sarthakpati** [reply]
I am unsure what you mean by this. GaNDLF is agnostic to the underlying environment/system/hardware as long as it supports PyTorch and Python. You will need to follow [the installation guide](https://mlcommons.github.io/GaNDLF/setup/) and customize it according to your setup. 

We have had multiple interns and tutorial participants use GaNDLF in Google Colab and GitHub Codespaces with tremendous success.

**Reply by MaimoonaAkram** [reply]
That's great.
I have installed conda, pytorch in google colab but when after create command when I write "!conda activate venv_gandlf"  command it gives me the following result:
![Uploading image.png…]()

**Reply by sarthakpati** [reply]
I am unable to see the error message. Can you re-upload, or paste the message [as formatted text](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks)?

**Reply by MaimoonaAkram** [reply]
The error text is given below:

**CommandNotFoundError: Your shell has not been properly configured to use 'conda activate'.
To initialize your shell, run

    $ conda init <SHELL_NAME>

Currently supported shells are:
  - bash
  - fish
  - tcsh
  - xonsh
  - zsh
  - powershell

See 'conda init --help' for more information and options.

IMPORTANT: You may need to close and restart your shell after running 'conda init'.**

**Reply by sarthakpati** [reply]
A quick internet search pointed me to this: https://stackoverflow.com/questions/61915607/commandnotfounderror-your-shell-has-not-been-properly-configured-to-use-conda
